app/views/WishList/index.tsx

```diff
@@ -64,25 +67,22 @@ interface Book {
     authors: Author[];
     image: Image;
     title: string;
-    mutate: (id: number) => void;
+    removeBookFromWishList: (id: string) => void;
+    wishListId: string;
 }
 
 function WishListBook(props: Book) {
-    const { id, price, authors, image, title, mutate } = props;
+    const { id, price, authors, image, title, removeBookFromWishList, wishListId } = props;
     const authorsDisplay = React.useMemo(() => (
         authors?.map((d) => d.name).join(', ')
     ), [authors]);
 
-    const removeBook = (_id: number) => {
-        mutate(_id);
-    };
+    const removeBook = useCallback(() => {
+        removeBookFromWishList(wishListId);
+    }, [wishListId]);
```

Comment:
add wishListId as name in button and directly set removeBookFromWishList in button's onclick